app/components/provider_interface/change_course_details_component.rb

```diff
@@ -0,0 +1,51 @@
+module ProviderInterface
+  class ChangeCourseDetailsComponent < ViewComponent::Base
```

Comment:
instead of copying over the component, can we leverage inheritance? (would be good to refer to the `ReadOnlyCompletedOfferSummaryComponent` -> `CompletedOfferSummaryComponent` -> `OfferSummaryComponent` structure) After that could we also tests to this component and a preview?